tests/bluetooth/bsim_bt/bsim_test_host/src/main_eatt.c

```diff
@@ -0,0 +1,491 @@
+	}
+
+	while (!is_connected) {
+		k_sleep(K_MSEC(100));
+	}
+
+	/* Wait a bit to ensure that all LLCP have time to finish */
+	k_sleep(K_MSEC(1000));
+
+	/* Wait for a while to disconnect */
+	k_sleep(K_MSEC(100000));
+
+	/* Disconnect */
+	err = bt_conn_disconnect(default_conn, BT_HCI_ERR_REMOTE_USER_TERM_CONN);
+	if (err) {
+		FAIL("Disconnection failed (err %d)\n", err);
+	}
+
+	while (is_connected) {
+		k_sleep(K_MSEC(100));
+	}
+
+	PASS("EATT Peripheral tests Passed\n");
+}
+
+static uint8_t discover_char_2_func(struct bt_conn *conn, const struct bt_gatt_attr *attr,
+				    struct bt_gatt_discover_params *params)
+{
+	// int err;
+
+	if (!attr) {
+		printk("Discover complete\n");
+		(void)memset(params, 0, sizeof(*params));
+		return BT_GATT_ITER_STOP;
+	}
+
+	char_2_attr_handle = attr->handle;
+
+	all_attributes_found = true;
+	return BT_GATT_ITER_STOP;
+}
+
+static uint8_t discover_char_1_func(struct bt_conn *conn, const struct bt_gatt_attr *attr,
+				    struct bt_gatt_discover_params *params)
+{
+	int err;
+
+	if (!attr) {
+		printk("Discover complete\n");
+		(void)memset(params, 0, sizeof(*params));
+		return BT_GATT_ITER_STOP;
+	}
+
+	char_1_attr_handle = attr->handle;
+
+	discover_params_2.uuid = &uuid_char_2.uuid;
+	discover_params_2.start_handle = service_handle + 1;
+	discover_params_2.type = BT_GATT_DISCOVER_ATTRIBUTE;
+	discover_params_2.end_handle = BT_ATT_LAST_ATTRIBUTE_HANDLE;
+	discover_params_2.func = discover_char_2_func;
+	discover_params_2.bearer_option = BT_ATT_BEARER_ANY;
+
+	err = bt_gatt_discover(conn, &discover_params_2);
+	return BT_GATT_ITER_STOP;
+}
+
+static uint8_t discover_primary_handler_func(struct bt_conn *conn, const struct bt_gatt_attr *attr,
+					     struct bt_gatt_discover_params *params)
+{
+	int err;
+
+	if (!attr) {
+		printk("Discover complete\n");
+		(void)memset(params, 0, sizeof(*params));
+		return BT_GATT_ITER_STOP;
+	}
+
+	service_handle = attr->handle + 1;
+
+	discover_params_2.uuid = &uuid_char_1.uuid;
+	discover_params_2.start_handle = attr->handle + 1;
+	discover_params_2.type = BT_GATT_DISCOVER_ATTRIBUTE;
+	discover_params_2.end_handle = BT_ATT_LAST_ATTRIBUTE_HANDLE;
+	discover_params_2.func = discover_char_1_func;
+	discover_params_2.bearer_option = BT_ATT_BEARER_ANY;
+
+	err = bt_gatt_discover(conn, &discover_params_2);
+	if (err) {
+		FAIL("Discover failed (err %d)\n", err);
+	}
+
+	return BT_GATT_ITER_STOP;
+}
+static void device_found(const bt_addr_le_t *addr, int8_t rssi, uint8_t type,
+			 struct net_buf_simple *ad)
+{
+	struct bt_le_conn_param *param;
+	int err;
+
+	err = bt_le_scan_stop();
+	if (err) {
+		FAIL("Stop LE scan failed (err %d)\n", err);
+	}
+
+	param = BT_LE_CONN_PARAM_DEFAULT;
+	err = bt_conn_le_create(addr, BT_CONN_LE_CREATE_CONN, param, &default_conn);
+
+	if (err) {
+		FAIL("Create conn failed (err %d)\n", err);
+	}
+	printk("Device connected\n");
+}
+
+uint8_t gatt_read_1_cb(struct bt_conn *conn, uint8_t err, struct bt_gatt_read_params *params,
```

Comment:
why you need 2 cb read functions exactly the same (duplicated) gatt_read_1_cd and gatt_read_2_cd?

Reply:
They just do different logging. There is probably more that can be simplified in the test.